src/main/java/jenkins/plugins/git/GitToolChooser.java

```diff
@@ -106,6 +136,150 @@ private void decideAndUseAPI(String remoteName, Item context, String credentials
         }
     }
 
+    /* Git repository URLs frequently end with the ".git" suffix.
+     * However, many repositories (especially https) do not require the ".git" suffix.
+     *
+     * Add remoteURL with the ".git" suffix and without the ".git" suffix to the
+     * list of alternatives.
+     */
+    private void addSuffixVariants(@NonNull String remoteURL, @NonNull Set<String> alternatives) {
+        alternatives.add(remoteURL);
+        String suffix = ".git";
+        if (remoteURL.endsWith(suffix)) {
+            alternatives.add(remoteURL.substring(0, remoteURL.length() - suffix.length()));
+        } else {
+            alternatives.add(remoteURL + suffix);
+        }
+    }
+
+    /* Git repository URLs frequently end with the ".git" suffix.
+     * However, many repositories (especially https) do not require the ".git" suffix.
+     *
+     * Add remoteURL with the ".git" suffix if not present
+     */
+    private String addSuffixVariant(@NonNull String canonicalURL) {
```

Comment:
I was initially perplexed by the name of the method.  Would you consider naming it `addSuffix` or `addSuffixIfMissing`?
```suggestion
    private String addSuffixVariant(@NonNull String canonicalURL) {
```